store/index.js

```diff
@@ -248,9 +274,50 @@ export const actions = {
     account = normalizeAddr(account);
     if (!state.activeAccount) commit('setAccount', account);
     if (state.accounts[account] === true) return; // Already added
-    state.accounts[account] = true;
+    this._vm.$set(state.accounts, account, true);
     commit('addAccount', account);
-  }
+  },
+
+  async detectHalfWrapped({ state, commit }, { ketherContract, nftContract, numBlocks }) {
+    // FIXME: Probably should move this into Wrap or a ResumeWrap component, no need to pollute global state
+    const account = state.activeAccount;
+    console.log("statrt")
+    if (!account) {
+      console.error("Can't detect half-wrapped ads without an active account. Connect a wallet.");
+      return;
+    }
+    if (numBlocks === undefined) {
+      numBlocks = 10 * 60 * 24 * 30; // Look 30 days back
+    }
+    let ids = {};
+    const eventFilter = [ ketherContract.filters.SetAdOwner() ];
+    const events = await ketherContract.queryFilter(eventFilter, -numBlocks);
+    for (const evt of events.reverse()) {
+      // Only look at events from your account (alas not indexed so we can't filter above)
+      if (normalizeAddr(evt.args.from) !== normalizeAddr(account)) continue;
+
+      const idx = evt.args.idx.toNumber();
+      if (ids[idx] !== undefined) continue;
+      ids[idx] = true;
+    }
+
+    for (const idx in ids) {
+      // Skip ads already wrapped successfully
+      if (state.ads[idx].wrapped) continue;
+     // if (state.ownedAds[idx]) continue;
```

Comment:
Intentional?

Reply:
If you half-wrap an ad by rejecting the transaction it will still be in ownedAds. We can also take it out in the first wrapping tx then put it back in....

Reply:
Not having it here gives a bunch of false positives, adding it back in.